docker/Dockerfile

```diff
@@ -121,41 +121,25 @@ RUN poetry build
 
 ################################ Stage: dependencies-dev (intermediate build target)
 
-FROM dependencies as dependencies-dev-platform-amd64
+FROM dependencies as dependencies-dev
+ARG TARGETARCH
 # Install hadolint for linting Dockerfiles
-RUN curl -Lo /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.10.0/hadolint-Linux-x86_64 && \
-    chmod +x /usr/bin/hadolint
+RUN if [ "$TARGETARCH" = "amd64" ]; then curl -Lo /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.10.0/hadolint-Linux-x86_64 && \
+    chmod +x /usr/bin/hadolint; fi
+RUN if [ "$TARGETARCH" = "arm64" ]; then curl -Lo /usr/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.10.0/hadolint-Linux-arm64 && \
+    chmod +x /usr/bin/hadolint; fi
 
 # Install NodeJS and use it to install markdownlint-cli
 
-RUN curl -Lo /tmp/node.tar.xz https://nodejs.org/dist/v17.9.0/node-v17.9.0-linux-x64.tar.xz && \
+RUN curl -Lo /tmp/node.tar.xz https://nodejs.org/dist/v17.9.0/node-v17.9.0-linux-$TARGETARCH.tar.gz && \
```

Comment:
Did you mean to swap from .xz to .gz ?

Reply:
This won't work because again NodeJS publishes `linux-x64` and `TARGETARCH` is `amd64`.

Reply:
https://github.com/nautobot/nautobot/runs/6100186033?check_suite_focus=true#step:10:257